lib/config/time.go

```diff
@@ -12,78 +12,86 @@ import (
 	"strings"
 )
 
-type Schedule struct {
-	Entries []ScheduleEntry `json:"entry" xml:"entry"`
+type Schedules struct {
+	RatesSchedule RatesSchedule `json:"ratesSchedule" xml:"ratesSchedule"`
 }
 
-type ScheduleEntry struct {
+type RatesSchedule struct {
+	Time        TimeFrame `json:"time" xml:"time"`
+	MaxSendKbps int       `json:"maxSendKbps" xml:"maxSendKbps"`
+	MaxRecvKbps int       `json:"maxRecvKbps" xml:"maxRecvKbps"`
+}
+
+type TimeFrame struct {
 	StartHour   int `json:"startHour" xml:"startHour"`
 	StartMinute int `json:"startMinute" xml:"startMinute"`
-	StartDay    int `json:"startDay" xml:"startDay"`
 	EndHour     int `json:"endHour" xml:"endHour"`
 	EndMinute   int `json:"endMinute" xml:"endMinute"`
-	EndDay      int `json:"endDay" xml:"endDay"`
-	MaxSendKbps int `json:"maxSendKbps" xml:"maxSendKbps"`
-	MaxRecvKbps int `json:"maxRecvKbps" xml:"maxRecvKbps"`
 }
 
-func ParseSchedule(s string) (Schedule, error) {
+func ParseSchedules(s string) (Schedules, error) {
```

Comment:
Not really sure if this is needed anymore.

I think the special parsing is only needed for TimeFrame (00:00-00:00, or perhaps break it into 2 x TimeMinute which parses just 00:00) but to be honest, given this now lives in it's own struct, perhaps it's ok to leave it as it is without any special parsing.

Also, I don't think we need any defaults here, all zeroes here are sensible defaults.

Reply:
Changed